Build basic React App with scaffold of what components to use.

jpwigdor opened the pull request.

jpwigdor committed:
Build basic React App with scaffold of what components to use.

jpwigdor committed:
Add side nav Route and some CSS

jpwigdor committed:
Add Functional Components to build Notes

jpwigdor committed:
Add additional components for use

jpwigdor committed:
Created ToDo list of how i'd like to approach the App.

Rewatching API request lectures before next commit

abravebee commented on lambda-notes/src/App.js:
```diff
@@ -0,0 +1,43 @@
+import './App.css';
+import { notEqual } from 'assert';
+
+//TODO: 
```
Some of these things seem pretty familiar! Remember a lot of this app is close to what you've built in the past, in terms of mapping through data and displaying a list, or linking to different pages in a SPA. Don't be afraid to self-plagiarize if it helps you! The key difference will be in how you work with the responses from the API, but this isn't as scary as it seems; you can use multiple calls to get the data you need, or look into lifecycle methods like componentDidUpdate(). And let me know if I can help you!

abravebee commented on lambda-notes/src/App.js:
```diff
@@ -0,0 +1,43 @@
+import NavBar from "./Components/NavBar";
+
+import './App.css';
+import { notEqual } from 'assert';
```
What are you planning to use this for?

jpwigdor replied:
it looks like something auto-populated and automatically imported that. It's been removed in my latest push!

abravebee commented on lambda-notes/src/App.js:
```diff
@@ -0,0 +1,43 @@
+        {/* <NewNote Component here - have note title field and note content /> */}
+        {/* <ViewNote Component here - edit/delete function included /> */}
+        {/* <EditNote Component here - edit/delete function included /> */}
+        {/* <DeleteNote Component here - This may not need to be a component. Perhaps a JS func with a popup box? /> */}
```
A pop up modal is definitely a good way to solve this problem!

abravebee commented on lambda-notes/src/Components/NoteList.js:
```diff
@@ -0,0 +1,11 @@
+const NoteList = () => {
+  return(
+    <div>
+      NoteList
```
If you're feeling stuck on the API calls, you can always start by working with dummy data and/or pseudocode what you'll be doing once the data is pulled in. For example, you probably know you'll need a map here, so you can note that if it helps you with forward momentum.

abravebee reviewed (commented):
I understand you're reviewing some API calls lectures--please let me know if you have any questions or I can help at all!!

jpwigdor committed:
Add Axios and pull API

built some base code to pull the note and populate in the notes component.
The mapping is running into an undefined error

jpwigdor committed:
Created more code to try and have cards show

made FETCH a function so the componentDidMount would simply call that, to look cleaner.

jpwigdor committed:
Edit Nav Bar to be more presentational

removed handler function for now to fix soon.

jpwigdor committed:
built NoteLists

but currently experiancing error with closing tags with JSX

jpwigdor committed:
NotesList complete

but error with map. So something must be wrong with my fetch method.

jpwigdor committed:
Still trying to work on making Map work properly

jpwigdor committed:
Sprint Submission
